docs: create how to use ipfs tutorial (#1207)
cosmos/sdk-tutorials: tutorials/how-to-use-ipfs/index.md

First, create an account on [Web3.storage](https://web3.storage/) and get an [API token](https://web3.storage/tokens/?create=true). With Web3.storage, you need to request access to the pinning API, or you can upload the file directly from the interface.

Next, add `web3.storage` as a remote service:

```sh
$ ipfs pin remote service add web3.storage https://api.web3.storage/ <YOUR_AUTH_KEY_JWT>
```

Make sure your node can serve the file when the Web3.storage server looks for it:

```sh
$ ipfs daemon &
```

Now you can pin the file to the remote node:

```sh
$ ipfs pin remote add --service=web3.storage --name="proposal-1-metadata.json" Qmbct5cv1SxMpEYCpNEvZnQ5TkN1rk7KyTmV95nbsrbLiR
```

Your file is now pinned to the remote node.

## 🎉 Congratulations 🎉

By completing this tutorial, you have learned how to use IPFS for storing files, notably for proposal metadata.

To learn more about IPFS in general, you can check their [documentation](https://docs.ipfs.io/).

> Additionally, you can learn more about how to use the `CID` in the [group](https://tutorials.cosmos.network/tutorials/understanding-group) and the [gov](https://tutorials.cosmos.network/tutorials/understanding-gov) modules tutorials.

Reviewer: @julienrbrt This understanding-gov is a broken link. Do you know where there is a current version?

For Cosmos chain governance, we want to store the metadata on IPFS nodes.
Let's create a file called `metadata.json` with the following content:

```json
{
  "title": "Funds alumni program",
  "authors": "IDA Alumni",
  "summary": "Lorem ipseum dolor sit amet",
  "details": "",
  "proposalForumURL": "https://forum.cosmos.network/t/foo",
  "voteOptionContext": "",
}
```

Add the file to your IPFS node:

```bash
$ ipfs add metadata.json
added Qmbct5cv1SxMpEYCpNEvZnQ5TkN1rk7KyTmV95nbsrbLiR metadata.json
```

By running `add` you have [pinned the file](https://docs.ipfs.tech/how-to/pin-files/#three-kinds-of-pins) to your node, stopping it from being deleted.
The content identifier (CID) of the file is `Qmbct5cv1SxMpEYCpNEvZnQ5TkN1rk7KyTmV95nbsrbLiR`. This is the CID that you will use to retrieve the file from IPFS.

Verify the content of the file by running:

```bash
$ ipfs cat Qmbct5cv1SxMpEYCpNEvZnQ5TkN1rk7KyTmV95nbsrbLiR
```

At this point, we can start our local IPFS node:

```bash
$ ipfs deamon
```

The file will become available on IPFS. We can easily access it via `https://ipfs.io/ipfs/<cid>`, in our case `https://ipfs.io/ipfs/Qmbct5cv1SxMpEYCpNEvZnQ5TkN1rk7KyTmV95nbsrbLiR`.

> Note that as soon as you stop the node, it is possible that after a while, it won't be accessible on the network.
> This is because only your node has the file being pinned, other nodes can delete the file depending of their retention policy.

Reviewer: ```suggestion
Note that as soon as you stop the node, it is possible that after a while, the file will not be accessible on the network.
This is because only your node has pinned the file, other nodes may delete the file depending of their retention policy.
```

> As we want to store the metadata permanently, without running our IPFS own node (for the scope of this tutorial), we want to pin it to a remote node, using a [pinning service](https://docs.ipfs.tech/how-to/pin-files/#three-kinds-of-pins).

Reviewer: ```suggestion
As you want to store the metadata permanently, without running your IPFS own node (for the scope of this tutorial), you want to pin it to a remote node, using a [pinning service](https://docs.ipfs.tech/how-to/pin-files/#three-kinds-of-pins).
```

## Pinning services

Pinning services are services that pin files on their nodes, making them available on the network, without having to run your own node.

> For this tutorial, we will use [Web3.storage](https://web3.storage/), a pinning service provided by **Protocol Labs**, the main developers of IPFS.

Reviewer: ```suggestion
For this tutorial, you will use [Web3.storage](https://web3.storage/), a pinning service provided by **Protocol Labs**, the main developers of IPFS.
```

Feel free to use any other pinning service you prefer, such as [Pinata](https://pinata.cloud), [Infura](https://infura.io/product/ipfs), etc.

First, create an account on [Web3.storage](https://web3.storage/) and get an [API token](https://web3.storage/tokens/?create=true).

> With [Web3.storage], you need to request access to the pinning API, or you can upload the file directly from the interface.

Reviewer: ```suggestion
With Web3.storage, you need to request access to the pinning API, or you can upload the file directly from the interface.
```

> Then let's add `web3.storage` as a remote service:

Reviewer: ```suggestion
Then add `web3.storage` as a remote service:
```

```bash
ipfs pin remote service add web3.storage https://api.web3.storage/ <YOUR_AUTH_KEY_JWT>
```

Now pin the file to the remote node:

```bash

> ipfs daemon & # we want to be sure the CID is available

Reviewer: ```suggestion
ipfs daemon & # you want to be sure the CID is available on the network
```

ipfs pin remote add --service=web3.storage --name="proposal-1-metadata.json" Qmbct5cv1SxMpEYCpNEvZnQ5TkN1rk7KyTmV95nbsrbLiR
```

> Our file is now pinned to the remote node.

Reviewer: ```suggestion
Your file is now pinned to the remote node.
```

## 🎉 Congratulations 🎉

By completing this tutorial, you have learned how to use IPFS for storing files, notably for proposal metadata.

> For learning more about IPFS in general, you can check their [documentation](https://docs.ipfs.io/).

Reviewer: ```suggestion
To learn more about IPFS in general, you can check their [documentation](https://docs.ipfs.io/).
```
